java/src/main/java/ai/rapids/cudf/ColumnVector.java

```diff
@@ -24,18 +24,99 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import java.util.ArrayList;
+import java.util.List;
 import java.util.Optional;
 import java.util.function.Consumer;
 
+import static ai.rapids.cudf.HostColumnVector.OFFSET_SIZE;
+
 /**
  * This class represents the immutable vector of data.  This class holds
  * references to device(GPU) memory and is reference counted to know when to release it.  Call
  * close to decrement the reference count when you are done with the column, and call incRefCount
  * to increment the reference count.
  */
-public final class ColumnVector implements AutoCloseable, BinaryOperable {
+public final class ColumnVector implements AutoCloseable, BinaryOperable, ColumnViewPointerAccess<BaseDeviceMemoryBuffer> {
   private static final Logger log = LoggerFactory.getLogger(ColumnVector.class);
 
+  public class DeviceColumnViewPointerAccess implements ColumnViewPointerAccess<BaseDeviceMemoryBuffer> {
+
+    protected long viewHandle;
+
+    public DeviceColumnViewPointerAccess(long viewHandle) {
+      this.viewHandle = viewHandle;
+    }
+
+    @Override
+    public long getColumnView() {
+      return viewHandle;
+    }
+
+    @Override
+    public ColumnViewPointerAccess<BaseDeviceMemoryBuffer> getChildColumnView(int childIndex) {
+      if (getDataType() != DType.LIST) {
+        return null;
+      }
+      long childColumnView = getChildCvPointer(viewHandle, childIndex);
+      //this is returning a new ColumnView - must close this!
+      return new DeviceColumnViewPointerAccess(childColumnView);
+    }
+
+    /**
+     * Gets the data buffer for the current column view (viewHandle).
+     * If the type is LIST it returns null.
+     * @return    If the type is LIST or data buffer is empty it returns null,
+     *            else return the data device buffer
+     */
+    @Override
+    public BaseDeviceMemoryBuffer getDataBuffer() {
+      long[] values = getNativeDataPointer(viewHandle);
+      if (values[0] == 0) {
+        return null;
+      }
+      return new DeviceMemoryBufferView(values[0], values[1]);
+    }
+
+    @Override
+    public BaseDeviceMemoryBuffer getOffsetBuffer() {
+      return offHeap.getNativeOffsetsPointer(viewHandle);
+    }
+
+    @Override
+    public BaseDeviceMemoryBuffer getValidityBuffer() {
+      return offHeap.getNativeValidPointer(viewHandle);
+    }
+
+    @Override
+    public long getNullCount() {
+      return  offHeap.getNativeNullCount();
+    }
+
+    @Override
+    public DType getDataType() {
+      return offHeap.getNativeType(viewHandle);
+    }
+
+    @Override
+    public long getNumRows() {
+      return offHeap.getNativeRowCount(viewHandle);
+    }
+
+    @Override
+    public int getNumChildren() {
+      if (getDataType() != DType.LIST /* or structs */) {
```

Comment:
what about strings? it has two children I believe

Reply:
Here, this method, if u look at the jni call as well, gives u number of `child`s of a column view.(excluding offsets since we dont quite need them thru this call)
strings - number of child columns = 0
list - number of child columns = 1
struct - number of child columns  >= 1

Reply:
> struct - number of child columns >= 1

`>=0`. :]

Reply:
Sure :). That's true with any column type really. I was trying to demonstrate what this could return.  num_children should indicate that isn't it?